activemq-broker/src/main/java/org/apache/activemq/broker/region/Queue.java

```diff
@@ -957,6 +957,11 @@ private void checkUsage(ConnectionContext context,ProducerBrokerExchange produce
     }
 
     private void expireMessages() {
+        if(isDispatchPaused()) {
```

Comment:
QueueDispatchSelector is final which is fine, specifically the paused flag inside that class isn't volatile because it's usually accessed under the consumer write lock everywhere

Reply:
Good catch. I'll update